model-optimizer/mo/front/onnx/loader.py

```diff
@@ -85,7 +85,9 @@ def protobuf2nx(graph, pb):
     for outp in pb.graph.output:
         name = str(outp.name)
         if graph.has_node(name):
-            raise Error('Name {} of output node already exists in graph.', name)
+            log.error('Name {} of output node already exists in graph. Ignoring this output. If the output is required,'
```

Comment:
I think that this check was here to check for cases when there is a **node** and a **tensor** with the same name. So raising the error in this case is still correct.

So, I believe that the better approach will be to find all unique values in the `pb.graph.output` and iterate over them instead of `pb.graph.output` directly.

Reply:
No, actually this check happens before nodes extraction, so it is safe to ignore the error here, it happens after inputs and constants extraction though.